src/homework8/arraylist/ArrayList.java

```diff
@@ -0,0 +1,119 @@
+package homework8.arraylist;
+
+import java.util.Arrays;
+
+public class ArrayList<T> implements List<T> {
+    private int position = 0;
+    private boolean checkIsEmpty;
+    private T[] array;
+
+    public ArrayList() {
+        checkIsEmpty = true;
+        array = (T[]) new Object[2];
+    }
+
+    @Override
+    public void add(T value) {
+        checkIsEmpty = false;
+        if (position < array.length) {
+            array[position] = value;
+            position++;
+        } else {
+            int newArrayLength = array.length + (array.length >> 1);
+            array = Arrays.copyOf(array, newArrayLength);
+            array[position] = value;
+            position++;
+        }
+    }
+
+    @Override
+    public void add(T value, int index) {
+        checkIsEmpty = false;
+        if (index < array.length && index > -1) {
+            array[index] = value;
```

Comment:
list = new ArrayList();
list. add(2, 2);
test what happens here?
This code should not work correctly)